mysql/mysql_gtid.go

```diff
@@ -234,11 +235,15 @@ func (s *UUIDSet) MinusInterval(in IntervalSlice) {
 	i, j := 0, 0
 	var minuend Interval
 	var subtrahend Interval
-	for j < len(in) && i < len(s.Intervals) {
+	for i < len(s.Intervals) {
 		if minuend.Stop != s.Intervals[i].Stop { // `i` changed?
 			minuend = s.Intervals[i]
 		}
-		subtrahend = in[j]
+		if j < len(in) {
+			subtrahend = in[j]
+		} else {
+			subtrahend = Interval{math.MaxInt64, math.MaxInt64}
```

Comment:
I think we can
```
append minuend to n
check if i+1 out of range and append s.Intervals[i+1:]... to n
```

it's OK not to change

Reply:
There were 2 bugs in 50 lines :(
So, I would prefer not to increase cyclomatic complexity of this code and keep it as it in PR.

Reply:
wait @atercattus 's review as this function is complex

Reply:
Hi, @atercattus have you had a chance to look into this?

Reply:
(I'm going to merge this PR if he does not respond it tomorrow)